AK/GenericLexer.h

```diff
@@ -71,6 +72,10 @@ class GenericLexer {
     }
 
     template<typename T>
+#ifndef KERNEL
+    // Pass a DeprecatedString by its StringView instead.
+    requires(!SameAs<T, DeprecatedString>)
```

Comment:
why? what's the point?

Reply:
The point was to give a better error message when people use C strings directly for this function. But I can remove it, the error will just be unreadable and non-actionable.

Reply:
Why does a C string even select the DeprecatedString overload instead of the StringView one?

Reply:
> Why does a string literal even select the DeprecatedString overload instead of the StringView one?

There's no conversion operator for string literals to StringView, we intentionally removed that for various reasons, the suffix operator isn't a conversion operator

> (In any case, wouldn't `= delete`-ing the overload do the trick?)

Good idea, will do.

Reply:
> There's no conversion operator for string literals to StringView, we intentionally removed that for various reasons.

That sounds... questionable?

Reply:
> That sounds... questionable?

According to #14555 the reasons were:
- Avoiding potentially broken implicit `strlen` calls
- Avoiding `strlen` calls on strings whose length is known at compile time

Reply:
@BertalanD the `StringView(char const*)` ctor was removed, and a templated ctor for string literals would be selected after `DeprecatedString(char const*)`.

Still, I'm not sure why using a C string is a bad thing in this case?